editor/src/app/sites/sites-state/sites.state.ts

```diff
@@ -37,94 +45,120 @@ export class SitesState implements NgxsOnInit {
+            this.store.dispatch(new RenameSiteSettingsSitenameAction(action.site, response.value));
+            this.store.dispatch(new RenameSiteTemplateSettingsSitenameAction(action.site, response.value));
+            this.store.dispatch(new RenameSectionTagsSitenameAction(action.site, response.value));
+            this.store.dispatch(new RenameSectionEntriesSitenameAction(action.site, response.value));
+          }
         }
-        return site;
-      })
-    );
+      });
   }
 
 
   @Action(RenameSiteAction)
   renameSite({ setState, getState }: StateContext<SiteStateModel[]>, action: RenameSiteAction) {
     const currentState = getState();
-
-    // @todo sync with backend, validate and return from server
     this.store.dispatch(new UpdateSiteAction(action.site, 'name', action.value));
-
-    this.store.dispatch(new RenameSiteSectionsSitenameAction(action.site, action.value));
-    this.store.dispatch(new RenameSiteSettingsSitenameAction(action.site, action.value));
-    this.store.dispatch(new RenameSiteTemplateSettingsSitenameAction(action.site, action.value));
-    this.store.dispatch(new RenameSectionTagsSitenameAction(action.site, action.value));
-    this.store.dispatch(new RenameSectionEntriesSitenameAction(action.site, action.value));
   }
 
 
   @Action(CloneSiteAction)
   cloneSite({ setState, getState }: StateContext<SiteStateModel[]>, action: CloneSiteAction) {
-    const currentState = getState();
-    const newSite: SiteStateModel = {
-      // @todo sync with backend
-      // @todo get new site from backend
-      // @todo sync related site data
-      name: 'clone-of-' + action.site.name,
-      title: 'Clone of ' + action.site.title,
-      order: currentState.length,
-      '@attributes': {
-        published: 0
-      }
-    };
-
-    setState(
-      [...currentState, newSite]
-    );
+    this.store.dispatch(new CreateSiteAction(action.site));
   }
 
 
   @Action(DeleteSiteAction)
-  DeleteSite({setState, getState}: StateContext<SiteStateModel[]>, action: DeleteSiteAction) {
-    const currentState = getState();
-    // @todo sync with backend
-    setState(
-      currentState
-        .filter(site => site.name !== action.site.name)
-        // Update order
-        .map((site, order) => {
-            return set('order', order, site);
-        })
-    );
-
-    this.store.dispatch(new DeleteSiteSectionsAction(action.site.name));
-    this.store.dispatch(new DeleteSiteSettingsAction(action.site.name));
-    this.store.dispatch(new DeleteSiteTemplateSettingsAction(action.site.name));
-    this.store.dispatch(new DeleteSiteSectionsTagsAction(action.site.name));
-    this.store.dispatch(new DeleteSiteSectionsEntriesAction(action.site.name));
+  DeleteSite({ setState, getState }: StateContext<SiteStateModel[]>, action: DeleteSiteAction) {
+    this.appStateService.sync('sites', { site: action.site.name }, 'DELETE')
+      .then((response: any) => {
```

Comment:
Let's keep to observables since we're already using them everywhere. So return observable from `sync` (remove the `toPromise`). And add `.pipe(map(response => {...}))` here.